### PR TITLE
UI Editor for picture-entity card

### LINKED ISSUE #3699: UI Editor: Picture Entity

### PR DESCRIPTION
**Pull request by iantrich**
Closes https://github.com/home-assistant/home-assistant-polymer/issues/3699

### CODE REVIEW

File: src/panels/lovelace/cards/hui-picture-entity-card.ts [hunk 1]
```diff
@@ -20,14 +20,32 @@ import computeStateName from "../../../common/entity/compute_state_name";
...
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import(/* webpackChunkName: "hui-picture-entity-card-editor" */ "../editor/config-elements/hui-picture-entity-card-editor");
+    return document.createElement("hui-picture-entity-card-editor");
+  }
+  public static getStubConfig(): object {
+    return {
+      entity: "",
+      image:
+        "https://www.home-assistant.io/images/merchandise/shirt-frontpage.png",
+      camera_view: "auto",
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-picture-entity-card.ts]: Same here, let's remove all optional config

File: src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+          .configValue="${"name"}"
+          @value-changed="${this._valueChanged}"
+        ></paper-input>
+        <paper-input
+          label="Image Url"
+          .value="${this._image}"
+          .configValue="${"image"}"
+          @value-changed="${this._valueChanged}"
+        ></paper-input>
+        <paper-input
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts]: Let's use a `entity-picker`

File: src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+          </paper-dropdown-menu>
+          <paper-input
+            label="Aspect Ratio"
+            type="number"
+            .value="${Number(this._aspect_ratio.replace("%", ""))}"
+            .configValue="${"aspect_ratio"}"
+            @value-changed="${this._valueChanged}"
+          ></paper-input>
+        </div>
+        <h3>Toggle Editor to input State Image</h3>
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts]: Does this mean the user has to enter the `yaml` editor to set state image up?

File: src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+import { ActionConfig } from "../../../../data/lovelace";
+import { PictureEntityCardConfig } from "../../cards/types";
+
+const cardConfigStruct = struct({
+  type: "string",
+  entity: "string",
+  image: "string?",
+  camera_image: "string?",
+  camera_view: "string?",
+  state_iamge: "object?",
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts]: The ui editor doesn't support `state_image`, so if the config contains `state_image` it should fallback to `yaml`.

Comment (iantrich) [reply src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts]: it should also be spelled correctly 😉

File: src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts (hunk 3 above)
Comment (iantrich) [reply src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts]: correct, I haven't built that out yet. Was going to create a generic state-image editor

Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts]: Not sure if we should show this message, normally we just don't show the function if we don't support it in the editor yet. I don't think this makes it much clearer what to do.

Comment (iantrich) [reply src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts]: I can get behind that

File: src/translations/en.json [hunk 5]
```diff
@@ -1113,6 +1113,12 @@
           },
           "theme": {
             "theme": "Theme (Optional)"
+          },
+          "card": {
```
Comment (iantrich) [inline src/translations/en.json]: thoughts on this translation strategy? could this be simpler? would add this for each config option

Comment (bramkragten) [reply src/translations/en.json]: You could create a separate key for optional and required, so we don't have to translate every key twice.

```json
{
"entity": "Entity",
"required": "Required",
"optional": "Optional"
}
```
```js
`${this.hass.localize("ui.panel.lovelace.editor.card.entity")} (${this.hass.localize("ui.panel.lovelace.editor.required")})`
```

**Comment by iantrich** [comment]
merging so I can update other editors

**Comment by iantrich** [comment]
I done goofed... :man_facepalming:

**Pull request merged**